src/device/rng.rs

```diff
@@ -0,0 +1,56 @@
+//! Driver for VirtIO random number generator devices.
+use crate::{queue::VirtQueue, transport::Transport, Hal, Result};
+
+// VirtioRNG only uses one queue
+const QUEUE_IDX: u16 = 0;
+const QUEUE_SIZE: usize = 8;
+
+bitflags::bitflags! {
+    #[derive(Clone, Copy, Debug, Default)]
+    struct RngFeature: u64 {}
+}
+
+/// Driver for a VirtIO random number generator device.
+pub struct VirtIORng<H: Hal, T: Transport> {
+    transport: T,
+    queue: VirtQueue<H, QUEUE_SIZE>,
+}
+
+impl<H: Hal, T: Transport> VirtIORng<H, T> {
+    /// Create a new driver with the given transport.
+    pub fn new(mut transport: T) -> Result<Self> {
+        transport.begin_init(RngFeature::empty());
```

Comment:
It should be simple to add support for the common `RING_EVENT_IDX` and `RING_INDIRECT_DESC` features. See `VirtIOInput::new` for an example of how to negotiate them and configure the queue appropriately.